From 2f2c9373519e000d717b0348650ea5df7705bc35 Mon Sep 17 00:00:00 2001
From: Christopher Degawa <ccom@randomderp.com>
Date: Thu, 29 Jun 2023 20:13:52 -0500
Subject: [PATCH] queue: disable qsorting if on Windows

qsort on Windows simply will cause the output of --static --libs to be
incorrect due to the way it sorts, and we have no easy way to work
around it.

Signed-off-by: Christopher Degawa <ccom@randomderp.com>
---
 libpkgconf/queue.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libpkgconf/queue.c b/libpkgconf/queue.c
index d7c768b..42b9164 100644
--- a/libpkgconf/queue.c
+++ b/libpkgconf/queue.c
@@ -204,7 +204,11 @@ next:
 	if (deps == NULL)
 		return;
 
+#ifndef _WIN32
 	qsort(deps, dep_count, sizeof (void *), dep_sort_cmp);
+#else
+	(void) dep_sort_cmp;
+#endif
 
 	/* zero the list and start readding */
 	pkgconf_list_zero(list);
-- 
2.41.0

